services/wallet/api.go

```diff
@@ -317,7 +317,12 @@ func (api *API) GetEthereumChains(ctx context.Context, onlyEnabled bool) ([]*par
 
 func (api *API) FetchPrices(ctx context.Context, symbols []string, currencies []string) (map[string]map[string]float64, error) {
 	log.Debug("call to FetchPrices")
-	return fetchCryptoComparePrices(symbols, currencies)
+	return api.s.priceManager.FetchPrices(symbols, currencies)
+}
+
+func (api *API) GetCachedPrices(ctx context.Context) (map[string]map[string]float64, error) {
+	log.Debug("call to GetCachedPrices")
+	return api.s.priceManager.GetCachedPrices()
 }
```

Comment:
How does a client knows when he has to call fetch prices or get cached prices.

Could we not make it transparent for the client?

Reply:
Hmm I guess it depends on whether you care or not about the values being the current ones. Whether you need fresh prices or old ones should be up to the client to decide.

In this particular case, the cache is used to have "some" price instead of none/having to fetch it during startup.